.github/workflows/release.yml

```
@@ -127,13 +122,52 @@ jobs:
 
       - uses: actions/checkout@v4
         with:
-          ref: ${{ inputs.branch_specifier || 'main' }}
           token: ${{ steps.get_token.outputs.token }}
 
       - uses: elastic/oblt-actions/git/setup@v1
         with:
           github-token: ${{ steps.get_token.outputs.token }}
 
-      - uses: ./.github/actions/setup
+      - name: Setting variables
+        id: variables
+        run: |
+          version=$(cat gradle.properties | grep -Po "(?<=version=)\d+\.\d+\.\d+")
+          { 
+            echo "version=$version"
+            echo "post_release_branch=post-release/$version"
+          } >> "$GITHUB_OUTPUT"
+
+      - if: ${{ ! inputs.dry_run }}
+        name: Creating GitHub release
+        run: |
+          release_tag=v$VERSION
+          git tag $release_tag
+          git push origin tag $release_tag
+          
+          version_without_dots=$(echo $VERSION | sed 's/\.//g')
+          release_notes="[Release Notes for $VERSION](https://www.elastic.co/guide/en/apm/agent/android/current/release-notes-1.x.html#elastic-apm-android-agent-$version_without_dots-release-notes)"
+          
+          gh release create $release_tag --title "Release $VERSION" --notes "$release_notes"
+        env:
+          VERSION: ${{ steps.variables.outputs.version }}
+          GH_TOKEN: ${{ steps.get_token.outputs.token }}
+
       - if: ${{ ! inputs.dry_run }}
-        run: ./gradlew postDeploy -Prelease=true -Pversion_override=${{ inputs.version_override_specifier || '' }}
+        name: Prepare for the next release
+        run: |
+          echo "Creating branch $POST_RELEASE_BRANCH"
+          git checkout -b $POST_RELEASE_BRANCH
+          
+          echo "Bumping version"
+          sed -i "s,version=$VERSION,version=$(./.github/scripts/bump-minor-version/main.sh $VERSION)," gradle.properties
+          git commit -a -m "Bumping version"
+          
+          echo "Pushing changes"
+          git push origin $POST_RELEASE_BRANCH
+          
+          echo "Creating PR"
+          gh pr create --title "Prepare for the next release" --base main --reviewer elastic/apm-agent-android --repo elastic/apm-agent-android
```

You can use `${{ github.repository }}` if it helps:
- https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#github-context

> `github.repository` | `string` | The owner and repository name. For example, `octocat/Hello-World`.


```suggestion
          gh pr create --title "Prepare for the next release" --base main --reviewer elastic/apm-agent-android --repo ${{ github.repository }}
```